src/flb_input_chunk.c

```diff
@@ -1495,6 +1495,15 @@ static int input_chunk_append_raw(struct flb_input_instance *in,
         pre_real_size = flb_input_chunk_get_real_size(ic);
     }
 
+    /*
+     * Set the total_records based on the records that n_records
+     * says we should be writing. This may be modified by flb_filter_do,
+     * where a filter may add/remove records.
```

Comment:
This only works because `flb_filter_do` **overwrites** the total records and not adds on it right? Maybe we drop a comment about that?

Saying that cause I had to stare at this change for too long to uderstand how this move avoids the double write

Reply:
Done in [8543d12](https://github.com/fluent/fluent-bit/pull/8229/commits/8543d126d287217567af15d16fdd0e42a56c71a3)